### PR TITLE
ledger: modernize

### PR DESCRIPTION
**Pull request by afh**
* Remove obsolete require "formula"
* Rename debug option to with-debug
* Use %w instead of %W
* Add with-check option, will change to without-check
  once tests pass reliably

### CODE REVIEW

File: Library/Formula/ledger.rb [hunk 1]
```diff
@@ -20,7 +18,8 @@ class Ledger < Formula
 
   head "https://github.com/ledger/ledger.git"
 
-  option "debug", "Build with debugging symbols enabled"
+  option "with-debug", "Build with debugging symbols enabled"
```
Comment (mikemcquaid) [inline Library/Formula/ledger.rb]: Needs a `deprecated_option` to migrate this.

File: Library/Formula/ledger.rb [hunk 2]
```diff
@@ -20,7 +18,8 @@ class Ledger < Formula
...
+  option "with-check", "Run build-time check"
```
Comment (mikemcquaid) [inline Library/Formula/ledger.rb]: Let's do this by default and add an option to turn it off (unless it takes a long time)

Comment (afh) [inline Library/Formula/ledger.rb]: Currently `acprep check` will fail sometimes, I'm working on fixing the issue upstream and I'd prefer to change it once the tests pass reliably.

File: Library/Formula/ledger.rb (hunk 1 above)
Comment (afh) [inline Library/Formula/ledger.rb]: Done.

**Comment by afh** [comment]
By the way have I mentioned that you guys are doing an excellent job?! if not: Y'all are making an excellent job :)

File: Library/Formula/ledger.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/ledger.rb]: Ok, let's omit the option for now then, thanks!

**Comment by mikemcquaid** [comment]
@afh Thanks (although we are not all guys; folks will do nicely :smile:)

**Comment by afh** [comment]
Sorry my bad I'm not a native speaker and for me guys includes all sexes and I should've known better; @mistydemeo has been very helpful in providing valuable feedback on my changes to Homebrew.

**Comment by mikemcquaid** [comment]
@afh Don't worry about it, just a small nit-pick :smiley:

File: Library/Formula/ledger.rb [hunk 3]
```diff
@@ -68,7 +68,9 @@ def install
 
     args += opts
 
+    system "./acprep", flavor, "make", *args
     system "./acprep", flavor, "make", "doc", *args
+    system "./acprep", flavor, "make", "check", *args if build.with? "check"
```
Comment (tdsmith) [inline Library/Formula/ledger.rb]: This needs an `option` declaration right?

Comment (afh) [inline Library/Formula/ledger.rb]: @mikemcquaid Requested that `make check` should be run and be turned off by the `without-check` option. Unfortunately `make check` randomly fails I'm working on a fix in the upstream repository, until then `make check` should therefore not be run at all and I've removed the line.

**Pull request closed without merging**